bitcoin/test/fuzz/miniscript.cpp

```diff
@@ -670,7 +670,11 @@ std::optional<NodeInfo> ConsumeNodeSmart(FuzzedDataProvider& provider, Type type
             while (subs.size() < children) subs.push_back(subs.back());
             return {{std::move(subs), frag, k}};
         }
+        default: break;
     }
+
+    assert(false);
+    return {};
```

Comment:
`default` and `return` should not be needed when using `assert(false)`, unless you are using a broken compiler, like mingw.

In fact `default` *disables* the warning you want to see?